api/src/components/publication/controller.ts

```diff
@@ -103,7 +104,13 @@ export const deletePublication = async (
             });
         }
 
-        if (publication.user.id !== event.user.id) {
+        const firstVersion = publication.versions.find((version) => version.versionNumber === 1);
```

Comment:
After discussing with PO about this flow, only the last version can be deleted (by it's creator) while it's still a DRAFT. If there's only one version, we delete the publication itself.

Reply:
Pushed a commit that made this a bit clearer - we can wait to write the code to delete a version later when we have a way to add more than one version.